gensim/models/phrases.py

```diff
@@ -98,7 +114,53 @@ def _is_single(obj):
         return False, obj_iter
 
 
-class Phrases(interfaces.TransformationABC):
+class SentenceAnalyzer:
+
+    def analyze_sentence(self, sentence, threshold, common_terms, scoring):
```

Comment:
You create a new class for avoiding copy-paste in two child classes, I'm correct?

Reply:
Yes that's it.